packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -311,26 +357,92 @@ List<Map<String, dynamic>> _extractPlatformMaps(List<Plugin> plugins, String typ
   return pluginConfigs;
 }
 
+/// Returns the version of the Android embedding that the current
+/// [project] is using.
+String _getAndroidEmbeddingVersion(FlutterProject project) {
+  if (!featureFlags.isNewAndroidEmbeddingEnabled) {
+    return '1';
+  }
+  assert(project.android != null);
+  final File androidManifest = project.android.appManifestFile;
+  assert(androidManifest.existsSync());
+  xml.XmlDocument document;
+  try {
+    document = xml.parse(androidManifest.readAsStringSync());
+  } on xml.XmlParserException {
+    throwToolExit('Error parsing ${project.android.appManifestFile} '
+                  'Please ensure that the android manifest is a valid XML document and try again.');
+  } on FileSystemException {
+    throwToolExit('Error reading ${project.android.appManifestFile} even though it exists. '
+                  'Please ensure that you have read permission to this file and try again.');
+  }
+  for (xml.XmlElement metaData in document.findAllElements('meta-data')) {
+    final String name = metaData.getAttribute('android:name');
+    if (name == 'flutterEmbedding') {
+      return metaData.getAttribute('android:value');
```

Comment:
It would probably be a good idea to check accepted values here and throw an appropriate error if a bad value is passed in.